workers/scheduler/scheduler.go

```diff
@@ -275,18 +283,59 @@ func (s *Scheduler) schedule() {
 			continue
 		}
 
-		// Mark them as scheduled
-		scheduled[id].Status = gaia.RunScheduled
+		// Check if this pipeline run is a docker run
+		if scheduled[id].Docker || gaia.Cfg.AutoDockerMode {
+			// Retrieve the global worker registration secret
+			if err = s.vault.LoadSecrets(); err != nil {
+				gaia.Cfg.Logger.Error("failed to load secrets from vault", "error", err)
+				continue
+			}
+			workerSecretBytes, err := s.vault.Get(gaia.WorkerRegisterKey)
+			if err != nil {
+				gaia.Cfg.Logger.Error("failed to get global worker registration secret from vault", "error", err)
+				continue
+			}
+			workerSecret := string(workerSecretBytes[:])
+
+			// Start docker worker for this pipeline run
+			worker := docker.NewDockerWorker(gaia.Cfg.DockerHostURL, scheduled[id].UniqueID)
+			if err := worker.SetupDockerWorker(gaia.Cfg.DockerRunImage, string(workerSecret[:])); err != nil {
+				gaia.Cfg.Logger.Error("failed to setup docker worker for pipeline run", "error", err)
+				continue
+			}
 
-		// Update entry in store
-		err = s.storeService.PipelinePutRun(scheduled[id])
-		if err != nil {
-			gaia.Cfg.Logger.Debug("could not put pipeline run into store", "error", err.Error())
+			// Cache docker worker
+			if err := s.memDBService.InsertDockerWorker(worker); err != nil {
+				gaia.Cfg.Logger.Error("failed to cache docker worker in memdb", "error", err)
+				if err := worker.KillDockerWorker(); err != nil {
+					gaia.Cfg.Logger.Error("failed to kill docker worker", "error", err)
```

Comment:
I feel like simply logging this fact is weird. At the least we should wrap it and pass it up, don't you think? Or provide some context, I don't know. If we just log it, I feel like we simply don't care and might as well not log it. Especially since Gaia is very verbose now and this might be just missed.

Reply:
> At the least we should wrap it and pass it up, don't you think?

Can you provide an example? If you mean with "pass it up" returning the error then this is not possible. We are in the schedule function which is periodically triggered and doesn't allow a return parameter.

The reason why I added this output is only for debugging purposes. What would you expect to happen here?

Reply:
Yeah the scheduler never returns an error. I guess it's okay, it's already in a failing statement.